bin/generate.py

```diff
@@ -1678,7 +1678,13 @@ def parse(key: str, name_gen: Callable[[], str], yaml: dict, parent: AnyDirector
 
             if is_testcase(yaml):
                 if isinstance(parent, RootDirectory):
-                    raise ParseException("Testcase must be inside Directory", name)
+                    raise ParseException("Test case must be inside a Directory.", name)
+
+                if not parent.numbered and key == "test_group":
+                    raise ParseException(
+                        "Test case must not be named 'test_group', this clashes with the group-level 'test_group.yaml'.",
+                        parent.path / name,
+                    )
 
```

Comment:
Can you move this into the `TestcaseRule`? I think the canonical place would be after checking that the name matches the `FILE_NAME_REGEX`.